pkg/stringset/stringset.go

```diff
@@ -26,3 +30,14 @@ func (ss StringSet) In(operand string) bool {
 	}
 	return false
 }
+
+func (ss StringSet) MatchRegex(operand string) bool {
+	for _, s := range ss {
+		match, _ := regexp.MatchString(s, operand)
```

Comment:
We need to handle the error in case the regex is malformed. I think the best way is to compile the regex at boot time using [regex.Compile()](https://pkg.go.dev/regexp#Compile). We can do so here in main go here after reading in the config file: https://github.com/spiffe/spire-controller-manager/blob/2b0ff48b84150cb1451234af111a0a2676134b3d/main.go#L104-L116

Step through ctrlConfig.IgnoreNamespaces and create a `[]regexp.Regex` and then pass this to the podReconciler instead of the array of strings:
https://github.com/spiffe/spire-controller-manager/blob/2b0ff48b84150cb1451234af111a0a2676134b3d/main.go#L291-L299

Then in `MatchRegex()` you can use [FindString()](https://pkg.go.dev/regexp#Regexp.FindString) which doesn't return err.

Reply:
If the regex is malformed, do we want to set ignoreNamespaces to the default value or only throw an error?

Reply:
Throw error